coordinator/src/node/rollover.rs

```diff
@@ -95,61 +89,131 @@ pub fn monitor(
+
+            let margin_coordinator = position.coordinator_margin.to_signed().expect("to fit");
+            let new_margin_coordinator = margin_coordinator - funding_fee;
+            let new_margin_coordinator = new_margin_coordinator.to_unsigned().expect("to fit");
+
+            let collateral_reserve_trader = collateral_reserve_trader.to_signed().expect("to fit");
+            let new_collateral_reserve_trader = collateral_reserve_trader + funding_fee;
+            let new_collateral_reserve_trader =
+                new_collateral_reserve_trader.to_unsigned().expect("to fit");
+
+            let new_leverage_coordinator =
+                calculate_leverage(quantity, new_margin_coordinator, average_entry_price)
+                    .expect("valid leverage");
+
+            let new_coordinator_liquidation_price = match position.trader_direction {
+                Direction::Long => calculate_short_liquidation_price(
+                    new_leverage_coordinator,
+                    average_entry_price,
+                    maintenance_margin_rate,
+                ),
+                Direction::Short => calculate_long_liquidation_price(
+                    new_leverage_coordinator,
+                    average_entry_price,
+                    maintenance_margin_rate,
+                ),
+            };
+
+            RolledOverPosition {
+                margin_coordinator: new_margin_coordinator,
+                margin_trader: position.trader_margin,
+                collateral_reserve_coordinator,
+                collateral_reserve_trader: new_collateral_reserve_trader,
+                leverage_coordinator: new_leverage_coordinator,
+                leverage_trader: decimal_from_f32(position.trader_leverage),
+                liquidation_price_coordinator: new_coordinator_liquidation_price,
+                liquidation_price_trader: decimal_from_f32(position.trader_liquidation_price),
+            }
+        }
+        FundingFee::TraderPays(funding_fee) => {
+            let funding_fee = funding_fee.to_signed().expect("to fit");
+
+            let margin_trader = position.trader_margin.to_signed().expect("to fit");
+            let new_margin_trader = margin_trader - funding_fee;
+            let new_margin_trader = new_margin_trader.to_unsigned().expect("to fit");
+
+            let collateral_reserve_coordinator =
+                collateral_reserve_coordinator.to_signed().expect("to fit");
+            let new_collateral_reserve_coordinator = collateral_reserve_coordinator + funding_fee;
+            let new_collateral_reserve_coordinator = new_collateral_reserve_coordinator
+                .to_unsigned()
+                .expect("to fit");
+
+            let new_leverage_trader =
+                calculate_leverage(quantity, new_margin_trader, average_entry_price)
+                    .expect("valid leverage");
+
+            let new_trader_liquidation_price = match position.trader_direction {
```

Comment:
I thought about it, but I thought it would not be needed. The position will get liquidated if it needs to be liquidated before or after the rollover anyway.

Reply:
I think it is: Liquidation is a timely topic: Let's just assume very high funding rates which would put your position margin below the maintenance margin requirements already on Monday, 00:00, so ~first funding rate event after the rollover. 
If we only liquidate the user on the next rollover event, we might not hedge said position for almost a week.

Reply:
Continue the discussion in this comment https://github.com/get10101/10101/pull/2532#discussion_r1599294462

Reply:
But, specifically to this code section, I don't think we need to handle it here. Here we are applying the already outstanding fees. The liquidation task should be considering this fees way before rollover is executed.

Reply:
Makes sense.